src/main/java/org/sunbird/progress/service/MandatoryContentServiceImpl.java

```diff
@@ -218,6 +222,78 @@ public Map<String, Object> getUserProgress(SunbirdApiRequest requestBody, String
 		return result;
 	}
 
+	public Map<String,Object> getBatchParticipantsByPage(Map<String, Object> request) {
+		Map<String,Object> responseMap = new HashMap<>();
+		Map<String, Object> queryMap = new HashMap<>();
+		queryMap.put(Constants.BATCH_ID, (String) request.get(Constants.BATCH_ID));
+		Map<String, Object> result = new HashMap<String, Object>();
+		List<String> userList = new ArrayList<String>();
+		Boolean active = (Boolean) request.get(Constants.ACTIVE);
+		if (null == active) {
+			active = true;
+		}
+		Integer limit = (Integer) request.get(Constants.LIMIT);
+		if (limit == null) {
+			limit = 10; // check default limit add it as constants
+		}
+		Integer currentOffSetFromRequest = (Integer) request.get(Constants.OFFSET);
+		if (currentOffSetFromRequest == null) {
+			currentOffSetFromRequest = 0;
+		}
+		String pageId = (String) request.get(Constants.PAGE_ID);
```

Comment:
This request object is created newly in getUserProgress method and it doesn't pass the pageId value... 
PageId must be sent back to UI and UI should return the value for next page read... 
If pageId is not in request body of which UI sent -- then backend should return the 1st page user details...